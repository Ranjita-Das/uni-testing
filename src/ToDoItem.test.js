import React from 'react';
import ToDo from './ToDoItem.js';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';


describe('to-do renders correctly', ()=>{

  it('displays the text based on props.text', ()=>{
    const wrapper = shallow (
      <ToDo
        text="this is the text"
        complete={false}
        deleteToDo={()=> null}
        toggleCompletion={()=>null}
      />
    );

    expect(wrapper.find('.to-do__info').text()).toBe("this is the text");
  });



   it('changes class based on props.complete', ()=>{
     const wrapperComplete = shallow(
       <ToDo
         text="this is the text"
         complete={true}
         deleteToDo={()=> null}
         toggleCompletion={()=>null}
       />
     )
     const wrapperIncomplete = shallow(
       <ToDo
         text="this is the text"
         complete={false}
         deleteToDo={()=> null}
         toggleCompletion={()=>null}
       />
     );
    expect(toJson(wrapperComplete)).toMatchSnapshot();
    expect(toJson(wrapperIncomplete)).toMatchSnapshot();

    expect(wrapperIncomplete.find('input').prop('complete')).toBeFalsy;
  });

  it('ToDo calls toggleCompletion on checkbox change', ()=>{

    const toggleCompletionSpy = jest.fn();
    const wrapper = shallow(
      <ToDo
        text="this is the text"
        complete={true}
        deleteToDo={()=> null}
        toggleCompletion={toggleCompletionSpy}
      />
      );

      wrapper.find('.to-do__completion').simulate('change');
      expect(toggleCompletionSpy).toHaveBeenCalled();
  });

  it('toDo calls DeleteToDo on clicking delete button', ()=>{
    const deleteToDoSpy = jest.fn();
    const wrapper = shallow(
      <ToDo
        text="this is the text"
        complete={true}
        deleteToDo={deleteToDoSpy}
        toggleCompletion={()=>null}
      />
      );

      wrapper.find('.button--delete').simulate('click');
      expect(deleteToDoSpy).toHaveBeenCalled();
  });

})
