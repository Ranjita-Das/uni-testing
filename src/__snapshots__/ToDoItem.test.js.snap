// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-do renders correctly changes class based on props.complete 1`] = `
<div
  className="to-do__item"
>
  <div
    className="to-do__info--complete"
  >
    <input
      checked={true}
      className="to-do__completion"
      onChange={[Function]}
      type="checkbox"
    />
    this is the text
  </div>
  <button
    className="button--delete"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;

exports[`to-do renders correctly changes class based on props.complete 2`] = `
<div
  className="to-do__item"
>
  <div
    className="to-do__info"
  >
    <input
      checked={false}
      className="to-do__completion"
      onChange={[Function]}
      type="checkbox"
    />
    this is the text
  </div>
  <button
    className="button--delete"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
